### PR TITLE
Add `fetchpriority="high"` when next/image has `priority` prop 📷

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Adding  `fetchpriority="high"` according to `priority` prop.

Inspired from Angular Image component.

_Sorry if I made any mistakes :(_

### CODE REVIEW

**Comment by sanjaiyan-dev** [comment]
Updated version of https://github.com/vercel/next.js/pull/40887/

**Comment by sanjaiyan-dev** [comment]
Extremely sorry please teach me to resolve merge conflict.

File: packages/next/client/image.tsx [hunk 1]
```diff
@@ -355,6 +355,64 @@ function handleLoading(
   })
 }
 
+sanjaiyan-image-loading-priority
```
Comment (styfle) [inline packages/next/client/image.tsx:358]: This looks like a bug

File: packages/next/client/image.tsx [hunk 2]
```diff
@@ -355,6 +355,63 @@ function handleLoading(
...
+  return (
+    <>
+      <img
+        {...rest}
+        {...imgAttributes}
+        width={widthInt}
+        height={heightInt}
+        decoding="async"
+        // https://web.dev/priority-hints/#increase-the-priority-of-the-lcp-image
+        fetchPriority={priority ? "high" : rest.fetchPriority}
```
Comment (styfle) [inline packages/next/client/image.tsx:393]: This will print an error with React

https://github.com/facebook/react/issues/25682

**Pull request closed without merging**